packages/store-ui/src/ProductQuantity/NumericStepper.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, Fragment, useCallback } from 'react'
+import { Button, Flex, Input } from 'theme-ui'
+
+export interface NumericStepperProps {
+  variant?: string
+  minValue: number | 1
+  maxValue?: number | undefined
+  onChange: (quantity: number) => void
+}
+
+export const NumericStepper: FC<NumericStepperProps> = ({
+  variant = 'productQuantity',
+  minValue,
+  maxValue,
+  onChange,
+  children,
+}) => {
+  var [quantity, setQuantity] = React.useState<number>(1)
+
+  const updateQuantity = useCallback(
+    (value: number) => {
+      setQuantity(value)
+      onChange(value)
+    },
+    [setQuantity, onChange]
+  )
+
+  const increaseNumber = useCallback(() => {
+    quantity = quantity + 1
+    if (maxValue) {
+      if (quantity > maxValue) quantity = maxValue
+    }
+    updateQuantity(quantity)
+  }, [quantity, maxValue])
+
+  const decreaseNumber = useCallback(() => {
+    quantity = quantity - 1
+    if (quantity < minValue) quantity = minValue
+    updateQuantity(quantity)
+  }, [quantity, minValue])
+
+  const setNumber = useCallback(
+    (value: number) => {
+      quantity = value
+      if (quantity < minValue) quantity = minValue
+      if (maxValue) {
+        if (quantity > maxValue) quantity = maxValue
+      }
+      updateQuantity(quantity)
+    },
+    [quantity, minValue, maxValue]
+  )
+
+  const isMax = maxValue ? quantity >= maxValue : false
+  const isMin = quantity <= minValue
+
+  return (
+    <Fragment>
+      <Flex variant={`${variant}.numericStepper`}>
+        <Button
+          variant={`${variant}.numericStepper.button.minus`}
+          onClick={decreaseNumber}
+          disabled={isMin}
+        >
+          -
+        </Button>
+        <Input
+          value={quantity}
+          onChange={(e) => setNumber(Number(e.target.value))}
```

Comment:
maybe should validate just in the onBlur event? makes sense?

Reply:
I think I understand, you mean when the user stops actually interacting with the input right?

Reply:
exactly, and you should call the ʻonChange` callback to, right? now just update the local `quantity` state, but don't notify the changes for the consumers of the component

Reply:
exactly, because now every key input calls the `onChange` callback